src/components/Button.tsx

```diff
@@ -45,30 +32,68 @@ const ThemedButton = styled.TouchableOpacity<ThemedButtonProps>`
   justify-content: center;
   height: ${hp('7%')}px;
   margin-bottom: ${hp('2%')}px;
+  background-color: ${props => props.backgroundColor};
+`
+
+const Label = styled.Text<ThemedLabelProps>`
+  color: ${props => props.color};
+  text-align: center;
+  font-family: ${props => props.theme.fonts.contentFontBold};
+  font-size: ${props => props.theme.fonts.defaultFontSize};
+  letter-spacing: ${props => props.theme.fonts.capsLetterSpacing};
+  text-transform: uppercase;
+  font-weight: ${props => props.theme.fonts.defaultFontWeight};
+  padding: 0 10px;
 `
 
-interface IButtonProps {
+interface ButtonPropsType {
   onPress: () => void
+  label: string
+  buttonTheme: ButtonThemeType
   disabled?: boolean
-  children: ReactNode
-  buttonTheme?: ButtonThemeType
+  iconLeft?: ComponentType<SvgProps>
+  iconRight?: ComponentType<SvgProps>
   testID?: string
 }
 
-const Button = ({ children, onPress, disabled = false, buttonTheme, testID }: IButtonProps): ReactElement => {
+const Button = (props: ButtonPropsType): ReactElement => {
   const [isPressed, setIsPressed] = React.useState(false)
+  const { label, onPress, disabled = false, buttonTheme = BUTTONS_THEME.light, testID } = props
+  let textColor = buttonTheme === BUTTONS_THEME.dark ? COLORS.lunesWhite : COLORS.lunesBlack
+  if (disabled) {
+    textColor = COLORS.lunesBlackLight
+  }
+
+  const getBackgroundColor = (): Color | 'transparent' => {
+    if (props.disabled) {
+      return COLORS.lunesBlackUltralight
+    }
+    if (isPressed) {
+      return props.buttonTheme === BUTTONS_THEME.dark ? COLORS.lunesBlackMedium : COLORS.lunesBlackLight
+    }
+    if (buttonTheme === 'dark') {
+      return COLORS.lunesBlack
+    } else {
+      return 'transparent'
+    }
+  }
+  const backgroundColor = getBackgroundColor()
+
   return (
     <ThemedButton
       buttonTheme={buttonTheme}
       testID={testID}
-      isPressed={isPressed}
+      backgroundColor={backgroundColor}
       onPress={onPress}
       disabled={disabled}
       onPressIn={() => setIsPressed(true)}
       onPressOut={() => setIsPressed(false)}
       activeOpacity={1}>
-      {children}
+      {props.iconLeft && <props.iconLeft fill={textColor} accessibilityLabel={'button-icon-left'} />}
```

Comment:
i didn't expect that you can do it that way ;)
i'm thinking about creating a separate ticket with a SVGIcon component. I think that could be the nicer way
What do you think? sth similar to that
https://michalmuszynski.com/blog/reusable-svg-component-in-react/

Reply:
yes, we can think about better handling of the SVGs, feel free to try this out, but passing the icon to a child (the button) should still work.